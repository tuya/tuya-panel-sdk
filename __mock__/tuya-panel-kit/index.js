import React from 'react';
import { View } from 'react-native';
import TYText from './components/TYText';
import UnitText from './components/UnitText';

module.exports = {
  TYText,
  UnitText,
  Utils: {
    RatioUtils: {
      convertX: () => {},
    },
  },
};
