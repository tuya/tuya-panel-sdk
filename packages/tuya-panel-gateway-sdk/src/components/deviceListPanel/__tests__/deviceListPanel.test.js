import React from 'react';
import { shallow } from 'enzyme';
import { Utils } from 'tuya-panel-kit';
import { Dimensions } from 'react-native';
import DeviceListPanel from '../index';
import Strings from '../i18n';

const { convertX: cx } = Utils.RatioUtils;
const { height: screenHeight } = Dimensions.get('screen');
const tabConfig = [
  {
    key: '0',
    title: Strings.getLang('bleDevices'),
    activeTextStyle: { color: '#00B996' },
    textStyle: { color: '#000' },
  },
  {
    key: '1',
    title: Strings.getLang('zigDevices'),
    activeTextStyle: { color: '#00B996' },
    textStyle: { color: '#000' },
  },
];
const devList = [];

describe('DeviceListPanel components', () => {
  it('basic render', () => {
    const wrapper = shallow(<DeviceListPanel tabs={tabConfig} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('change postion', () => {
    const wrapper = shallow(
      <DeviceListPanel tabs={tabConfig} initialPosition={screenHeight / 4} />
    );
    expect(wrapper).toMatchSnapshot();
  });
  // it('change prompt text and style', () => {
  //   const wrapper = shallow(
  //     <DeviceListPanel
  //       tabs={tabConfig}
  //       prompt="During the process of adding, the device will no longer be used, please be patient."
  //       promptStyle={{ color: 'lightblue' }}
  //     />
  //   );
  //   expect(wrapper).toMatchSnapshot();
  // });
  // it('change progress text and style', () => {
  //   const wrapper = shallow(
  //     <DeviceListPanel
  //       tabs={tabConfig}
  //       foreColor={{
  //         '0%': 'yellow',
  //         '100%': 'green',
  //       }}
  //       progressStyle={{ with: cx(100), height: cx(100) }}
  //       progressText="4 / 8"
  //     />
  //   );
  //   expect(wrapper).toMatchSnapshot();
  // });
  // it('change is custom progress change', () => {
  //   const wrapper = shallow(
  //     <DeviceListPanel isCustomProgressChange={true} customTotal={10} customProgress={4} />
  //   );
  //   expect(wrapper).toMatchSnapshot();
  // });
});
