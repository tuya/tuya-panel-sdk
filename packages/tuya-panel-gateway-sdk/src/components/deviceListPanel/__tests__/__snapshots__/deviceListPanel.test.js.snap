// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeviceListPanel components basic render 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <AnimatedComponent
    style={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "justifyContent": "flex-end",
          "position": "absolute",
          "width": "100%",
        },
        Object {
          "height": 88,
          "top": -88,
        },
      ]
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "height": 60,
          "justifyContent": "center",
          "marginLeft": 20,
          "width": 60,
        }
      }
    >
      <Component
        name="arrow"
        size={16}
        style={
          Object {
            "transform": Array [
              Object {
                "rotate": "90deg",
              },
            ],
          }
        }
      />
    </TouchableOpacity>
  </AnimatedComponent>
  <AnimatedComponent
    onLayout={[Function]}
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "#FAFAFA",
          "bottom": 0,
          "position": "absolute",
          "width": "100%",
        },
        undefined,
        Object {
          "borderRadius": 20,
          "top": 667,
        },
      ]
    }
  >
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "alignSelf": "center",
            "height": 112,
            "justifyContent": "space-between",
            "marginBottom": 20,
            "width": "100%",
          },
          Object {
            "paddingHorizontal": 48,
          },
        ]
      }
    >
      <TouchableOpacity
        activeOpacity={0.2}
        style={
          Array [
            Object {
              "alignSelf": "center",
              "height": 8,
              "marginTop": 10,
              "width": 54,
            },
            Object {
              "backgroundColor": "rgb(235,235,235)",
            },
          ]
        }
      />
      <_class2
        activeKey="0"
        defaultActiveKey={0}
        isUnderlineCenter={true}
        onChange={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "rgb(235,235,235)",
            },
            undefined,
          ]
        }
        tabActiveStyle={Object {}}
        tabActiveTextStyle={Object {}}
        tabStyle={Object {}}
        tabTextStyle={Object {}}
        tabs={
          Array [
            Object {
              "activeTextStyle": Object {
                "color": "#00B996",
              },
              "key": "0",
              "textStyle": Object {
                "color": "#000",
              },
              "title": "Bluetooth device list",
            },
            Object {
              "activeTextStyle": Object {
                "color": "#00B996",
              },
              "key": "1",
              "textStyle": Object {
                "color": "#000",
              },
              "title": "Zigbee device list",
            },
          ]
        }
        type="radio"
        underlineStyle={Object {}}
        wrapperStyle={Object {}}
      />
    </View>
    <FlatList
      columnWrapperStyle={
        Object {
          "justifyContent": "space-between",
        }
      }
      contentContainerStyle={
        Object {
          "flexGrow": 1,
          "paddingHorizontal": 48,
        }
      }
      data={Array []}
      disableVirtualization={false}
      horizontal={false}
      initialNumToRender={20}
      key="1"
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={2}
      onEndReachedThreshold={2}
      onScroll={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEnabled={false}
      scrollEventThrottle={50}
      showsVerticalScrollIndicator={false}
      style={
        Object {
          "flex": 1,
        }
      }
      updateCellsBatchingPeriod={50}
      windowSize={21}
    />
  </AnimatedComponent>
</View>
`;

exports[`DeviceListPanel components change postion 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <AnimatedComponent
    style={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "justifyContent": "flex-end",
          "position": "absolute",
          "width": "100%",
        },
        Object {
          "height": 88,
          "top": -88,
        },
      ]
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "height": 60,
          "justifyContent": "center",
          "marginLeft": 20,
          "width": 60,
        }
      }
    >
      <Component
        name="arrow"
        size={16}
        style={
          Object {
            "transform": Array [
              Object {
                "rotate": "90deg",
              },
            ],
          }
        }
      />
    </TouchableOpacity>
  </AnimatedComponent>
  <AnimatedComponent
    onLayout={[Function]}
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "#FAFAFA",
          "bottom": 0,
          "position": "absolute",
          "width": "100%",
        },
        undefined,
        Object {
          "borderRadius": 20,
          "top": 333.5,
        },
      ]
    }
  >
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "alignSelf": "center",
            "height": 112,
            "justifyContent": "space-between",
            "marginBottom": 20,
            "width": "100%",
          },
          Object {
            "paddingHorizontal": 48,
          },
        ]
      }
    >
      <TouchableOpacity
        activeOpacity={0.2}
        style={
          Array [
            Object {
              "alignSelf": "center",
              "height": 8,
              "marginTop": 10,
              "width": 54,
            },
            Object {
              "backgroundColor": "rgb(235,235,235)",
            },
          ]
        }
      />
      <_class2
        activeKey="0"
        defaultActiveKey={0}
        isUnderlineCenter={true}
        onChange={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "rgb(235,235,235)",
            },
            undefined,
          ]
        }
        tabActiveStyle={Object {}}
        tabActiveTextStyle={Object {}}
        tabStyle={Object {}}
        tabTextStyle={Object {}}
        tabs={
          Array [
            Object {
              "activeTextStyle": Object {
                "color": "#00B996",
              },
              "key": "0",
              "textStyle": Object {
                "color": "#000",
              },
              "title": "Bluetooth device list",
            },
            Object {
              "activeTextStyle": Object {
                "color": "#00B996",
              },
              "key": "1",
              "textStyle": Object {
                "color": "#000",
              },
              "title": "Zigbee device list",
            },
          ]
        }
        type="radio"
        underlineStyle={Object {}}
        wrapperStyle={Object {}}
      />
    </View>
    <FlatList
      columnWrapperStyle={
        Object {
          "justifyContent": "space-between",
        }
      }
      contentContainerStyle={
        Object {
          "flexGrow": 1,
          "paddingHorizontal": 48,
        }
      }
      data={Array []}
      disableVirtualization={false}
      horizontal={false}
      initialNumToRender={20}
      key="1"
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={2}
      onEndReachedThreshold={2}
      onScroll={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEnabled={false}
      scrollEventThrottle={50}
      showsVerticalScrollIndicator={false}
      style={
        Object {
          "flex": 1,
        }
      }
      updateCellsBatchingPeriod={50}
      windowSize={21}
    />
  </AnimatedComponent>
</View>
`;
